File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/skylanders/model/Skylander.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.skylanders.model;

public class Skylander
{
  private SkylanderPair mPair;
  private String mName;

  public final static Skylander BLANK_SKYLANDER = new Skylander(-1, -1, "Blank");

  public Skylander(int id, int var, String name)
  {
    mPair = new SkylanderPair(id, var);
    mName = name;
  }

  public String getName()
  {
    return mName;
  }

  public void setName(String name)
  {
    this.mName = name;
  }

  public int getId()
  {
    return mPair.getId();
  }

  public int getVar()
  {
    return mPair.getVar();
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/skylanders/model/Skylander.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.skylanders.model

class Skylander(id: Int, variant: Int, var name: String) {
    private val pair: SkylanderPair = SkylanderPair(id, variant)

    val id: Int get() = pair.id
    val variant: Int get() = pair.variant

    companion object {
        @JvmField
        val BLANK_SKYLANDER = Skylander(-1, -1, "Blank")
    }
}
